boards/arm/teensy4/teensy40.dts

```diff
@@ -35,15 +38,20 @@
 };
 
 &flexspi {
+	status = "okay";
+
 	reg = <0x402a8000 0x4000>, <0x60000000 0x200000>;
 	/* WINBOND flash memory*/
 	w25q16jvuxim: w25q16jvuxim@0 {
-		compatible = "winbond,w25q16jvuxim";
-		size = <16777208>;
+		compatible = "nxp,imx-flexspi-nor";
+		size = <16777216>;
```

Comment:
You could use `DT_SIZE_M(2 * 8)` here instead of just the number. Up to you really. Same for the teensy41 case (`DT_SIZE_M(8 * 8)`)

Reply:
For one I didn't knew these shortcuts exited. Are these evaluated during parsing time of the DTS? (Otherwise wouldn't the NXP Kconfig script which defines FLASH_SIZE have issues?)

Secondly I haven't yet put the effort into calculating wether the amount of blocks is actually equivalent to pure 8MB

Reply:
These are implemented as precompiler macros (`dts/common/mem.h`), so it would expand to `((2 * 8) * 1024 * 1024)`, then that would be fed to the compiler and it would evaluate it at build time, resulting in something statically initialized to `16777216`.

Reply:
thx for these suggestions. TIL. I've integrated them in the latest commit